packages/plugin-react/src/fast-refresh.ts

```diff
@@ -1,8 +1,11 @@
 import fs from 'node:fs'
 import path from 'node:path'
 import { createRequire } from 'node:module'
+import { type Options } from './index'
 
-export const runtimePublicPath = '/@react-refresh'
+export const runtimePublicPath = (
+  reactRefreshPrefix?: Options['reactRefreshPrefix'],
+): string => `${reactRefreshPrefix ?? ''}/@react-refresh`
```

Comment:
All usages without options can keep using the `runtimePublicPath`
Idem for usages in index.ts

You can inline this template string concat in the `sharedHeader` function